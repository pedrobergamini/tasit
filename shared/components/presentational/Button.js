import React from "react";
import { StyleSheet } from "react-native";
import RNButton from "react-native-button";
import Colors from "@constants/Colors";
import PropTypes from "prop-types";
import {
  responsiveHeight,
  responsiveFontSize,
} from "react-native-responsive-dimensions";

// Responsive percents
const FONT_SIZE = 1.8;
const BUTTON_HEIGHT = 1.5;

// TODO: Change to NativeBase button component
// https://github.com/tasitlabs/tasit/issues/204
export default function Button(props) {
  let { title, onPress, disabled } = props;
  if (disabled === undefined) disabled = false;

  const onPressHandler = disabled ? () => {} : onPress;

  title = title.toUpperCase();

  return (
    <RNButton
      containerStyle={
        disabled ? containerStyles.disabled : containerStyles.enabled
      }
      style={disabled ? buttonStyles.disabled : buttonStyles.enabled}
      onPress={onPressHandler}
      activeOpacity={1}
    >
      {title}
    </RNButton>
  );
}

Button.propTypes = {
  title: PropTypes.string.isRequired,
  onPress: PropTypes.func.isRequired,
  disabled: PropTypes.bool,
};

const containerStyles = StyleSheet.create({
  enabled: {
    padding: responsiveHeight(BUTTON_HEIGHT),
    overflow: "hidden",
    borderRadius: 3,
    backgroundColor: Colors.buttonBackground,
  },
  disabled: {
    padding: responsiveHeight(BUTTON_HEIGHT),
    overflow: "hidden",
    borderRadius: 3,
    backgroundColor: Colors.disabledButtonBackground,
  },
});

const buttonStyles = StyleSheet.create({
  disabled: {
    fontSize: responsiveFontSize(FONT_SIZE),
    color: Colors.disabledButtonColor,
  },
  enabled: {
    fontSize: responsiveFontSize(FONT_SIZE),
    color: Colors.buttonColor,
  },
});
