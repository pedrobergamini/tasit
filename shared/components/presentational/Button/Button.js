import React from "react";
import { StyleSheet } from "react-native";
import RNButton from "react-native-button";
import Colors from "@shared-constants/Colors";
import PropTypes from "prop-types";
import {
  responsiveHeight,
  responsiveFontSize,
} from "react-native-responsive-dimensions";

// Responsive percents
const FONT_SIZE = 1.8;
const BUTTON_HEIGHT = 1.5;

// TODO: Change to NativeBase button component
// https://github.com/tasitlabs/tasit/issues/204
export default function Button(props) {
  let { title, onPress, disabled } = props;
  if (disabled === undefined) disabled = false;

  const onPressHandler = disabled ? () => {} : onPress;

  title = title.toUpperCase();

  return (
    <RNButton
      containerStyle={
        disabled ? containerStyles.disabled : containerStyles.enabled
      }
      style={disabled ? buttonStyles.disabled : buttonStyles.enabled}
      onPress={onPressHandler}
      activeOpacity={1}
    >
      {title}
    </RNButton>
  );
}

Button.propTypes = {
  title: PropTypes.string.isRequired,
  onPress: PropTypes.func.isRequired,
  disabled: PropTypes.bool,
};

const containerStyles = StyleSheet.create({
  disabled: {
    backgroundColor: Colors.disabledButtonBackground,
    borderRadius: 3,
    overflow: "hidden",
    padding: responsiveHeight(BUTTON_HEIGHT),
  },
  enabled: {
    backgroundColor: Colors.buttonBackground,
    borderRadius: 3,
    overflow: "hidden",
    padding: responsiveHeight(BUTTON_HEIGHT),
  },
});

const buttonStyles = StyleSheet.create({
  disabled: {
    color: Colors.disabledButtonTextColor,
    fontSize: responsiveFontSize(FONT_SIZE),
  },
  enabled: {
    color: Colors.buttonTextColor,
    fontSize: responsiveFontSize(FONT_SIZE),
  },
});
