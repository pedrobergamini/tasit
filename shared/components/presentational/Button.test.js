import React from "react";
import { shallow } from "enzyme";
import Button from "shared/components/presentational/Button";

describe("Button", () => {
  describe("renders the component", () => {
    const onPress = () => {};

    it("default button", async () => {
      expect(
        shallow(<Button title="Enabled button" onPress={onPress} />)
      ).toMatchSnapshot();
    });

    it("disabled button", async () => {
      const disabled = true;
      expect(
        shallow(
          <Button
            title="Disabled button"
            onPress={onPress}
            disabled={disabled}
          />
        )
      ).toMatchSnapshot();
    });
  });
});
