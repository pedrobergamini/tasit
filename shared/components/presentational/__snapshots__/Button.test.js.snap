// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders the component default button 1`] = `
<Button
  activeOpacity={1}
  containerStyle={
    Object {
      "backgroundColor": "#07FFC2",
      "borderRadius": 3,
      "overflow": "hidden",
      "padding": 20.009999999999998,
    }
  }
  onPress={[Function]}
  style={
    Object {
      "color": "black",
      "fontSize": 27.53633962602873,
    }
  }
>
  ENABLED BUTTON
</Button>
`;

exports[`Button renders the component disabled button 1`] = `
<Button
  activeOpacity={1}
  containerStyle={
    Object {
      "backgroundColor": "gray",
      "borderRadius": 3,
      "overflow": "hidden",
      "padding": 20.009999999999998,
    }
  }
  onPress={[Function]}
  style={
    Object {
      "color": "white",
      "fontSize": 27.53633962602873,
    }
  }
>
  DISABLED BUTTON
</Button>
`;
